File: signing-consentor-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__SigningConsentorDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.oauth.consent.signing.SigningConsentor;
import se.curity.identityserver.sdk.plugin.descriptor.SigningConsentorPluginDescriptor;

public final class ${pluginName}SigningConsentorDescriptor implements SigningConsentorPluginDescriptor<${pluginName}ConsentorConfig> {

    @Override
    public Class<? extends SigningConsentor> getConsentorType()
    {
        return ${pluginName}SigningConsentor.class;
    }

    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}ConsentorConfig> getConfigurationType()
    {
        return ${pluginName}ConsentorConfig.class;
    }
}

File: kotlin-signing-consentor-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__SigningConsentorDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.SigningConsentorPluginDescriptor

class ${pluginName}SigningConsentorDescriptor: SigningConsentorPluginDescriptor<${pluginName}ConsentorConfig>
{
    override fun getConsentorType() = ${pluginName}SigningConsentor::class.java

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}ConsentorConfig::class.java
}
